### ISSUE TITLE
Process is terminated due to StackOverflowException.

### ISSUE THREAD

**Issue by basictheprogram** [ansible/ansible#10825]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 2.0.0 (devel 716a87ff36) last updated 2015/04/23 12:58:07 (GMT -500)
  lib/ansible/modules/core: (detached HEAD 022e5274e6) last updated 2015/01/11 15:27:34 (GMT -500)
  lib/ansible/modules/extras: (detached HEAD a32869d492) last updated 2015/01/11 15:27:43 (GMT -500)
  v2/ansible/modules/core: (detached HEAD 27d5e5124f) last updated 2015/01/11 15:27:48 (GMT -500)
  v2/ansible/modules/extras: (detached HEAD 8a4f07eecd) last updated 2015/01/11 15:27:53 (GMT -500)
  configured module search path = None
##### Environment:

Control: OS X 10.10.3
Client: Windows 7 Pro 64-bit
##### Summary:

Attempting to make changes to Local Administrator account and I get a StackOverflowException.
##### Steps To Reproduce:

```
- name: Clean up Local Administrator
  win_user: name=Administrator description=Local fullname=Local password_never_expires=yes
```

```
$ ansible-playbook --ask-pass -i windows-workstations windows-workstations.yml
TASK: [windows-workstations | Clean up Local Administrator] *******************
failed: [windows-test.local] => {"failed": true, "parsed": false}
```

Process is terminated due to StackOverflowException.

I did execute the play with -vvvv but I'm not sure what to post here. It's pretty long.
##### Expected Results:

Local Administrator would have the description and fullname set to Local and the password set to never expire.
##### Actual Results:

Process is terminated due to StackOverflowException.

**Comment by bcoca** [comment]
@cchurch

**Comment by basictheprogram** [comment]
I believe this is the relevant parts of the -vvvv. Let me know if you need more.

<windows-test.local> WINRM PUT /var/folders/d_/bm7rvz154jb_2djkqybb503h0000gn/T/tmpKfvknP to C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1429830677.31-160705540905978\arguments (offset=0 size=103)
<windows-test.local> EXEC PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -File C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1429830677.31-160705540905978\win_user.ps1 C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1429830677.31-160705540905978\arguments; Remove-Item "C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1429830677.31-160705540905978\" -Force -Recurse;
<windows-test.local> WINRM EXEC 'PowerShell' ['-NoProfile', '-NonInteractive', '-EncodedCommand', 'UABvA <snip>']
<windows-test.local> WINRM RESULT <Response code 0, out "", err " Process is terminat">
failed: [windows-test.local] => {"failed": true, "parsed": false}

**Comment by sivel** [comment]
iirc there is a windows hotfix that addresses this issues.

**Comment by sivel** [comment]
It may be http://support.microsoft.com/kb/2842230

I found that referenced on the mailing list

**Comment by basictheprogram** [comment]
That hotfix does resolve this issue.

**Comment by jhawkesworth** [comment]
We should document this.

.. note::
   Due to a bug in Windows Management Framework 3.0, it may be necessary to
   install http://support.microsoft.com/kb/2842230 to avoid receiving
   Out of memory and StackOverflowExceptions.

   Windows 8.1 and Server 2012 R2 are not affected by this issue as they
   come with Windows Management Framework 4.0.

(add the above to ansible/docsite/rst/intro_windows.rst in the windows
system prep section.

But .. having found this page

http:/
mikefrobbins.com/2015/01/08/hot-to-check-the-powershell-version-and-install-a-new-version/

I think it might be better advice to suggest updating to WMF 4.0
which is available from Windows 7 /Server 2008 R2 onwards.

This would leave behind users of Server 2008 - everything before that
(Vista, server 2003 don't even support powershell 3.0) -  ansible has never
promised to work with anything that doesn't support powershell 3.0 if i
recall.

**Comment by basictheprogram** [comment]
And/or change http://docs.ansible.com/intro_windows.html

Getting to PowerShell 4.0 or higher
PowerShell 4.0 or higher is needed for most provided Ansible modules for Windows, and is also required to run the above setup script. Note that PowerShell 4.0 is only supported on Windows 7 SP1, Windows Server 2008 R2, and later releases of Windows.

**Comment by basictheprogram** [comment]
> On Apr 24, 2015, at 12:59 AM, jhawkesworth notifications@github.com wrote:
> 
> http:/
> mikefrobbins.com/2015/01/08/hot-to-check-the-powershell-version-and-install-a-new-version/ http://mikefrobbins.com/2015/01/08/hot-to-check-the-powershell-version-and-install-a-new-version/
> 
> I think it might be better advice to suggest updating to WMF 4.0
> which is available from Windows 7 /Server 2008 R2 onwards.

Can confirm that WMF 4.0 on Windows 7 64-bit works.

## 

Bob Tanner tanner@real-time.com                       | Phone : 952-943-8700
http://www.real-time.com, Linux, VMware, Fortigate | Fax      : 952-943-8500
Key fingerprint = 9906 320A 8BB6 64AD 96A7  7785 CBFB 10BF 568B F98C

**Comment by jhawkesworth** [comment]
Really we should discuss this on the ansible project mailing list.

Do you want to raise it?

Jon

**Comment by cchurch** [comment]
I think this issue has been addressed by the documentation changes in #10926

**Comment by furiml** [comment]
I often get this error with large PS1 scripts, even with WMF 4 and WMF 5.
